csharp/src/sdk/Elements/Space.cs

```diff
@@ -0,0 +1,163 @@
+using Newtonsoft.Json;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Hypar.Geometry;
+
+namespace Hypar.Elements
+{
+    /// <summary>
+    /// A space represents an extruded boundary an occupiable region within a building.
+    /// </summary>
+    public class Space : Element, ILocateable<Polyline>, ITessellate<Mesh>, IMaterialize
+    {
+        private List<Polyline> _sides = new List<Polyline>();
+
+        /// <summary>
+        /// The bottom perimeter of the space.
+        /// </summary>
+        /// <returns></returns>
+        [JsonProperty("location")]
+        public Polyline Location{get;}
+
+        /// <summary>
+        /// The elevation of the bottom perimeter.
+        /// </summary>
+        [JsonProperty("bottom_elevation")]
+        public double BottomElevation{get;}
+
+        /// <summary>
+        /// The height of the space above its bottom elevation.
+        /// </summary>
+        [JsonProperty("height")]
+        public double Height{get;}
+
+        /// <summary>
+        /// The material of the Mass.
+        /// </summary>
+        /// <value></value>
+        [JsonIgnore]
+        public Material Material{get;set;}
+
+        /// <summary>
+        /// Construct a default space.
+        /// </summary>
+        public Space()
+        {
+            var defaultProfile = Profiles.Rectangular();
+            this.Location = defaultProfile;
+            this.BottomElevation = 0.0;
+            this.Height = 1.0;
+            this.Material = BuiltInMaterials.Default;
+        }
+
+        /// <summary>
+        /// Construct a space from a perimeter and a height.
+        /// </summary>
+        /// <param name="bottom">The bottom perimeter of the space.</param>
+        /// <param name="bottomElevation">The elevation of the bottom perimeter.</param>
+        /// <param name="height">The height of the space above the bottom elevation.</param>
+        /// <returns></returns>
+        public Space(Polyline bottom, double bottomElevation, double height)
+        {
+
+            if (height <= 0.0)
+            {
+                throw new ArgumentOutOfRangeException(Messages.HEIGHT_EXCEPTION, "height");
+            }
+
+            this.Location = bottom;
+            this.BottomElevation = bottomElevation;
+            this.Material = BuiltInMaterials.Default;
+        }
+
+        /// <summary>
+        /// A collection of curves representing the vertical edges of the space.
+        /// </summary>
+        /// <returns></returns>
+        public IEnumerable<Line> VerticalEdges()
```

Comment:
Is this functionality we need right now? Same question for `Faces()` and `VerticalEdges()`. If not, remove these.

Reply:
Nope, left them in just in case you thought they should be there.

Reply:
Removed HorizonalEdges, VerticalEdges, and Faces.